employee-rostering-frontend/src/ui/components/OptionalInput.tsx

```diff
@@ -67,6 +67,9 @@ export default class OptionalInput<T> extends React.Component<OptionalInputProps
                 this.props.onChange(mappedValue);
               }
             }
+            else {
+              this.props.onChange(undefined);
+            }
           }}
         />
         <Switch aria-label="Enabled" isChecked={isChecked} onChange={this.handleToggle} />
```

Comment:
This component isn't necessary, and it's use is extra work for users (they have to toggle the switch for every field they want to fill). Most (if not all) input components have `isRequired` property, see [patternfly's TextInput](https://www.patternfly.org/v4/documentation/react/components/textinput/#textinput-props).

Reply:
Patternfly Design Docs has two views on this: https://www.patternfly.org/v4/design-guidelines/usage-and-behavior/forms. The first is to mark required fields with an "*" or put a message "All fields are optional" by the form. The second is to have a checkbox that enables an field.

Reply:
So in the contract case, only the name is required and others are optional, so you can add `isRequired` to the name TextInput which will display an asterisk on top of it and leave the rest as TextInput without `isRequired`, they'll be optional by default.
I didn't quite get the second view, a required input isn't up to the user to specify, developers decide whether a field is required or not and force users to fill it if so.